### DISCUSSION TITLE
Metamask QR Code Scan not working properly in WalletConnectUnity

### DISCUSSION THREAD

**Discussion by yozdev** [WalletConnect/walletconnect-monorepo#2867] [Developer Support]
Hello Team,

I imported the latest assets from 2.0 branch of WalletConnectUnity git repo in my Unity project.

While running the SimpleUnityDapp Demo scene in the editor, I noticed that scanning the generated QR code does nothing. I don't receive any notification on my app nor any logs in the editor.

I also received missing scripts warning in the console when starting the scene. Also noticed some missing scripts references on the QR gameobjects.

I'm attaching screenshots of the warnings and QR gameobject.

Please guide me on how to use the QR code scanner in v2.

<img width="1920" alt="QR Missing Script" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/113970224/b3d5d3d3-5e8e-482e-b1b5-2ed08483b56e">

<img width="1920" alt="Missing scripts" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/113970224/69bd7468-4ece-4552-92dc-03d048505579">

**Comment by yozdev** [comment]
Would really appreciate a guide or some instructions on how to use QR code scanner in v2.

**Comment by finessevanes** [comment]
Hey there! Could we please get the following to help:
1) Version of the package you are using
2) Reproducible code via a codeBox, error logs, and/or reproduction steps
3) Which wallet (platform and version) you used to get to the issue

**Reply by yozdev** [reply]
Hey @finessevanes. Thanks for responding.

I imported the latest assets from https://github.com/WalletConnect/WalletConnectUnity 2.0 branch. I'm working on Unity Editor version 2021.3.23f1, Windows platform.

Following are the steps I followed :

1) I ran the Demo scene that came with the assets.
2) Script missing warnings show up immediately after the scene starts
3) I selected Ethereum Goerli chain and then pressed Connect button
4) QR code gets displayed
5) I scan the QR code using the latest Android Metamask app ver. 7.1.0
6) I receive the "Connect to this site" notification in my Metamask app
7) After pressing Connect, I get the "Connected to a simple unity demo..." message on Metamask app. However, no other logs are generated in Unity and the scene is stuck with the QR code displayed.
8) I checked the WalletConnect sessions in my Metamask app as well but no sessions were displayed there.

I used the Metamask wallet, Android version 7.1.0

**Comment by yozdev** [comment]
Hey @finessevanes

Is the Metamask bug being investigated ?

I tested with Trust Wallet and it's working fine.

Metamask does not return any response to Unity after successful QR scan.
